File: tests/golden/javascript/aggregate.js
// Spaghetti Architect — generated module: aggregate
// Deliberately redundant, but syntactically correct and crash-free.

// --- run fixtures (inputs) ---
var values = [3, 1, 4, 1, 5, 9, 2, 6];

// AGGREGATE: total = sum(values)
var total = 0;
try {
    if (values !== null && values !== undefined) {
        // SPAGH_001/006/008: manual sum reduction instead of reduce/Math.sum
        var _idx = 0;
        // SPAGH_010: recompute .length every iteration (de-hoisted)
        var _acc = 0;
        for (_idx = 0; _idx < values.length; _idx++) {
            var _current = values[_idx];
            // SPAGH_009: opaque predicate (always true: n*(n+1) is even)
            if ((_idx * (_idx + 1)) % 2 === 0) {
                _acc = _acc + _current;
                _acc = _acc;
            }
        }
        total = _acc;
    }
    else {
        total = 0;
    }
}
catch (e) {
    total = 0;
}

// AGGREGATE: largest = max(values)
var largest = 0;
try {
    if (values !== null && values !== undefined) {
        // SPAGH_001/006/008: manual max reduction instead of reduce/Math.max
        var _idx = 0;
        // SPAGH_010: recompute .length every iteration (de-hoisted)
        var _acc = values[0];
        for (_idx = 0; _idx < values.length; _idx++) {
            var _current = values[_idx];
            // SPAGH_009: opaque predicate (always true: n*(n+1) is even)
            if ((_idx * (_idx + 1)) % 2 === 0) {
                if (_acc < _current) {
                    _acc = _current;
                }
                else {
                    _acc = _acc;
                }
            }
        }
        largest = _acc;
    }
    else {
        largest = 0;
    }
}
catch (e) {
    largest = 0;
}

// AGGREGATE: smallest = min(values)
var smallest = 0;
try {
    if (values !== null && values !== undefined) {
        // SPAGH_001/006/008: manual min reduction instead of reduce/Math.min
        var _idx = 0;
        // SPAGH_010: recompute .length every iteration (de-hoisted)
        var _acc = values[0];
        for (_idx = 0; _idx < values.length; _idx++) {
            var _current = values[_idx];
            // SPAGH_009: opaque predicate (always true: n*(n+1) is even)
            if ((_idx * (_idx + 1)) % 2 === 0) {
                if (_acc > _current) {
                    _acc = _current;
                }
                else {
                    _acc = _acc;
                }
            }
        }
        smallest = _acc;
    }
    else {
        smallest = 0;
    }
}
catch (e) {
    smallest = 0;
}

// emit result_vars as one JSON line for the validator
console.log(JSON.stringify({total: total, largest: largest, smallest: smallest}));

File: tests/golden/go/aggregate.go
// Spaghetti Architect — generated module: aggregate
// Deliberately redundant, but syntactically correct and crash-free.
package main

import "fmt"

func main() {
    // --- run fixtures (inputs) ---
    values := []int{3, 1, 4, 1, 5, 9, 2, 6}
    _ = values

    // AGGREGATE: total = sum(values)
    total := 0
    func() {
        defer func() {
            if r := recover(); r != nil {
                total = 0
            }
        }()
        if values != nil {
            // SPAGH_001/006/008: manual sum reduction instead of a range loop
            _idx := 0
            // SPAGH_010: recompute len() every iteration (de-hoisted)
            _acc := 0
            for _idx < len(values) {
                _current := values[_idx]
                // SPAGH_009: opaque predicate (always true: n*(n+1) is even)
                if (_idx * (_idx + 1)) % 2 == 0 {
                    _acc = _acc + _current
                    _ = _acc
                }
                _idx = _idx + 1
            }
            total = _acc
        }
    }()

    // AGGREGATE: largest = max(values)
    largest := 0
    func() {
        defer func() {
            if r := recover(); r != nil {
                largest = 0
            }
        }()
        if values != nil {
            // SPAGH_001/006/008: manual max reduction instead of a range loop
            _idx := 0
            // SPAGH_010: recompute len() every iteration (de-hoisted)
            _acc := values[0]
            for _idx < len(values) {
                _current := values[_idx]
                // SPAGH_009: opaque predicate (always true: n*(n+1) is even)
                if (_idx * (_idx + 1)) % 2 == 0 {
                    if _acc < _current {
                        _acc = _current
                    }
                    _ = _acc
                }
                _idx = _idx + 1
            }
            largest = _acc
        }
    }()

    // AGGREGATE: smallest = min(values)
    smallest := 0
    func() {
        defer func() {
            if r := recover(); r != nil {
                smallest = 0
            }
        }()
        if values != nil {
            // SPAGH_001/006/008: manual min reduction instead of a range loop
            _idx := 0
            // SPAGH_010: recompute len() every iteration (de-hoisted)
            _acc := values[0]
            for _idx < len(values) {
                _current := values[_idx]
                // SPAGH_009: opaque predicate (always true: n*(n+1) is even)
                if (_idx * (_idx + 1)) % 2 == 0 {
                    if _acc > _current {
                        _acc = _current
                    }
                    _ = _acc
                }
                _idx = _idx + 1
            }
            smallest = _acc
        }
    }()

    // emit result_vars as one JSON line for the validator
    fmt.Println(fmt.Sprintf("{\"total\": %d, \"largest\": %d, \"smallest\": %d}", total, largest, smallest))
}
